### ISSUE TITLE
Unsupported video mode: `b' \x18\x00\xff\x00\xff\x00\xff\x00\xff\x10\x08\x00'`

### ISSUE THREAD

**Issue by goodboy** [barneygale/asyncvnc#2]
When running the following script against the following server command:
`x11vnc -display :0 -noipv6 -forever -noxdamage -ncache_cr`

target script:
```python
async def run_client():
    async with asyncvnc.connect(
        'localhost',
        port=5901,
    ) as client:
        print(client)

    asyncio.run(run_client())
```
output error:

```python
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/goodboy/repos/piker/snippets/async_vnc.py", line 8, in run_client
    async with asyncvnc.connect(
  File "/usr/lib/python3.10/contextlib.py", line 199, in __aenter__
    return await anext(self.gen)
  File "/home/goodboy/repos/asyncvnc/asyncvnc.py", line 563, in connect
    client = await Client.create(reader, writer, username, password, host_key)
  File "/home/goodboy/repos/asyncvnc/asyncvnc.py", line 514, in create
    video=await Video.create(reader, writer))
  File "/home/goodboy/repos/asyncvnc/asyncvnc.py", line 292, in create
    mode = video_modes[await reader.readexactly(13)]
KeyError: b' \x18\x00\xff\x00\xff\x00\xff\x00\xff\x10\x08\x00'
```

Feels like a full "system test" where you spawn `x11vnc` and run a client against it would be handy 😄

**Comment by goodboy** [comment]
I hacked in a `force_video_mode:str = 'rgba'` that gets passed on through to `Video.create()` and that seemed to make everything work. Not sure if that's the solution you want though 😂

**Comment by barneygale** [comment]
Try 45c311729b330f92ff92dee28ad9f18c0ea093ee?

**Comment by goodboy** [comment]
Ah nice, will do now.

**Comment by goodboy** [comment]
Yeah don't think it fixed it.

**Comment by barneygale** [comment]
That video mode uses 16 bits per pixel, which my code can't/won't support. It's different from your original exception!

**Comment by goodboy** [comment]
@barneygale yeah i haven't changed anything. you can see the command line i'm using.

I suggest you run the `x11vnc -display :0 -noipv6 -forever -noxdamage -ncache_cr` and try to run your code.

If you don't expect that to work it'd be somewhat odd to me,
- this is a super popular server setup on linux
- the code works fine if i just force `rgba` (as mentioned)

So i'm not sure what you'd like to do. I can put up my force mode PR and you can see it work?

**Comment by goodboy** [comment]
Ahh ok I take it back, the [latest error](https://github.com/barneygale/asyncvnc/issues/2#issuecomment-1135250766) is actually from an `x11vnc` running in docker setup - so you prolly are right about it being different (i think i connected to the wrong server by accident on that last manual test).

I'm putting up a small test suite shortly that shows your current code works on a local server.
I guess it might be worth it to also show a small container using the settings i have and showing that it works when forcing `'rgba'` mode even though you said:

> That video mode uses 16 bits per pixel, which my code can't/won't support.

**Comment by barneygale** [comment]
I'm not convinced that the X11 server tells you it's using 16-bit pixels, and then proceeds to send 32-bit pixels; such a thing is an obvious protocol error that would break any VNC client.

**Comment by goodboy** [comment]
> I'm not convinced that the X11 server tells you it's using 16-bit pixels, and then proceeds to send 32-bit pixels; such a thing is an obvious protocol error that would break any VNC client.

Yeah I have no clue, not my area of knowledge whatsoever 😂 
I'm just trying to get your nice client to work for automation.

@barneygale see the test suite in #4 - pretty sure the problem is still there on a plain old server even on https://github.com/barneygale/asyncvnc/commit/45c311729b330f92ff92dee28ad9f18c0ea093ee history.

**Comment by goodboy** [comment]
@barneygale for sure, but all you need to do is run that test suite to check 😉

**Comment by goodboy** [comment]
@barneygale 💥 

```python
tests/test_x11vnc.py::test_basic_connection_maybe_auth[password=None-force_vid=None] PASSED                                                             [ 25%]
tests/test_x11vnc.py::test_basic_connection_maybe_auth[password=None-force_vid=rgba] PASSED                                                             [ 50%]
tests/test_x11vnc.py::test_basic_connection_maybe_auth[password=doggy-force_vid=None] PASSED                                                            [ 75%]
tests/test_x11vnc.py::test_basic_connection_maybe_auth[password=doggy-force_vid=rgba] PASSED                                                            [100%]

====================================================================== 4 passed in 5.28s ======================================================================
```

Nice work friend!

**Comment by goodboy** [comment]
So I think getting that test suite in would be super duper slick 😉

**Comment by goodboy** [comment]
Yup resolved in https://github.com/barneygale/asyncvnc/commit/afbdf1d80ddd112d9c823b0d00fc8e666d592b34 🏄🏼
